src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs

```diff
@@ -1314,151 +1408,152 @@ private MessageImportance ChooseReferenceLoggingImportance(Reference reference)
         /// </summary>
         private void LogInputs()
         {
-            if (Traits.Instance.EscapeHatches.LogTaskInputs)
+            if (Traits.Instance.EscapeHatches.LogTaskInputs || Silent)
             {
                 // the inputs will be logged automatically anyway, avoid duplication in the logs
                 return;
             }
 
-            if (!Silent)
-            {
-                Log.LogMessageFromResources(MessageImportance.Low, "ResolveAssemblyReference.LogTaskPropertyFormat", "TargetFrameworkMoniker");
-                Log.LogMessageFromResources(MessageImportance.Low, "ResolveAssemblyReference.FourSpaceIndent", _targetedFrameworkMoniker);
+            string indent = Strings.FourSpaces;
+            string property = Strings.LogTaskPropertyFormat;
+            MessageImportance importance = MessageImportance.Low;
 
-                Log.LogMessageFromResources(MessageImportance.Low, "ResolveAssemblyReference.LogTaskPropertyFormat", "TargetFrameworkMonikerDisplayName");
-                Log.LogMessageFromResources(MessageImportance.Low, "ResolveAssemblyReference.FourSpaceIndent", TargetFrameworkMonikerDisplayName);
+            Log.LogMessage(importance, property, "TargetFrameworkMoniker");
```

Comment:
General question: I tried chasing when the string eventually gets formatted and stopped my search at `LoggingService.RouteBuildEvent`, the args get passed into a buildeventsink.Consume. Where do they get picked up and eventually formatted?

Reply:
When either SerialConsoleLogger or ParallelConsoleLogger actually wants to print it. If there are no text loggers, the string is never concatenated together. Well maybe in the binlog viewer after reading the binlog.

The beauty of LazyFormattedBuildEventArgs is that arguments travel separately from the string through the entire system. We haven't been leveraging this until now. Node packet translator already did the right thing, and now the binlog reader and writer do too.